crates/biome_css_analyze/src/lint/nursery/no_invalid_direction_in_linear_gradient.rs

```diff
@@ -0,0 +1,164 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_css_syntax::{CssFunction, CssParameter};
+use biome_rowan::AstNode;
+use biome_rowan::AstSeparatedList;
+use lazy_static::lazy_static;
+use regex::Regex;
+
+use crate::utils::vendor_prefixed;
+
+declare_rule! {
+    /// Disallow non-standard direction values for linear gradient functions.
+    ///
+    /// A valid and standard direction value is one of the following:
+    /// - an angle
+    /// - to plus a side-or-corner (`to top`, `to bottom`, `to left`, `to right`; `to top right`, `to right top`, `to bottom left`, etc.)
+    ///
+    /// A common mistake (matching outdated non-standard syntax) is to use just a side-or-corner without the preceding to.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```css,expect_diagnostic
+    /// .foo { background: linear-gradient(top, #fff, #000); }
+    /// ```
+    ///
+    /// ```css,expect_diagnostic
+    /// .foo { background: linear-gradient(45, #fff, #000); }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```css
+    /// .foo { background: linear-gradient(to top, #fff, #000); }
+    /// ```
+    ///
+    /// ```css
+    /// .foo { background: linear-gradient(45deg, #fff, #000); }
+    /// ```
+    ///
+    pub NoInvalidDirectionInLinearGradient {
+        version: "next",
+        name: "noInvalidDirectionInLinearGradient",
+        language: "css",
+        recommended: true,
+        sources: &[RuleSource::Stylelint("function-linear-gradient-no-nonstandard-direction")],
+    }
+}
+
+lazy_static! {
+    // It is necessary to find case-insensitive string.
+    // For example, both 'linear-gradinet' and 'Linear-gradient' should pass the check.
+    pub static ref LINEAR_GRADIENT_FUNCTION_NAME: Regex =
+        Regex::new(r"^(?i)(-webkit-|-moz-|-o-|-ms-)?linear-gradient").unwrap();
+
+    // It is necessary to find case-insensitive string.
+    // Also Check if 'in' is a word.
+    // For examples,`to top in srgb` is valid but `to top insrgb` is not valid.
+    pub static ref IN_KEYWORD: Regex = Regex::new(r"(?i)\bin\b").unwrap();
+
+    // This regex checks if a string consists of a number immediately followed by a unit, with no space between them.
+    pub static ref ANGLE: Regex = Regex::new(r"^[\d.]+(?:deg|grad|rad|turn)$").unwrap();
+
+    // It is necessary to find case-insensitive string.
+    // For example, both 'top' and 'TOP' should pass the check.
+    pub static ref DIRECTION: Regex = Regex::new(r"(?i)top|left|bottom|right").unwrap();
+
+    // This need for capture 'side-or-corner' keyword from linear-gradient function.
+    // Ensure starts with the keyword 'to' and ends with the keyword 'side-or-corner'.
+    pub static ref DIRECTION_WITH_TO: Regex = Regex::new(&format!(
+        r"(?i)^to ({})(?: ({}))?$",
+        DIRECTION.as_str(),
+        DIRECTION.as_str()
+    ))
+    .unwrap();
```

Comment:
Where's the word `side-or-corner`?

Reply:
I was referring to the `side-or-corner` keyword mentioned in the MDN documentation.
[mdn side-or-corner](https://developer.mozilla.org/en-US/docs/Web/CSS/gradient/linear-gradient#side-or-corner